frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -67,7 +67,7 @@ function ReaderPaging:registerKeyEvents()
         self.key_events.GotoNextPos = { { "Down" }, event = "GotoPosRel", args = 1, }
         self.key_events.GotoPrevPos = { { "Up" }, event = "GotoPosRel", args = -1, }
     end
-    if Device:hasKeyboard() then
+    if Device:hasKeyboard() and Device.model ~= "Kindle3" then
```

Comment:
Are there actually conflicting key binds?

Because otherwise I'd argue those are mildly useful. (Ditto for Rolling).

Reply:
It's conflicting in the sense that Alt+W (apparently turned into `2` at some level lower than us) is the Wikipedia shortcut. For consistently it would have to be either all or nothing.

But maybe there's some service that can be killed or massaged not to do thatinstead?

Reply:
It's a native evdev device, so I would assume that's done by the kernel (i.e., no, we can't do anything about it).

Reply:
Yup, it's the kernel:

```C
 237   │ /*!
 238   │  * For now, do mapping for top-row keys which are numeric when ALT is pressed.
 239   │  */
 240   │ unsigned int alt_convert(unsigned int keycode) {
 241   │     switch (keycode) {
 242   │     case KEY_Q:
 243   │        return KEY_1;
 244   │        break;
 245   │     case KEY_W:
 246   │        return KEY_2;
 247   │        break;
 248   │     case KEY_E:
 249   │        return KEY_3;
 250   │        break;
 251   │     case KEY_R:
 252   │        return KEY_4;
 253   │        break;
 254   │     case KEY_T:
 255   │        return KEY_5;
 256   │        break;
 257   │     case KEY_Y:
 258   │        return KEY_6;
 259   │        break;
 260   │     case KEY_U:
 261   │        return KEY_7;
 262   │        break;
 263   │     case KEY_I:
 264   │        return KEY_8;
 265   │        break;
 266   │     case KEY_O:
 267   │        return KEY_9;
 268   │        break;
 269   │     case KEY_P:
 270   │        return KEY_0;
 271   │        break;
 272   │     default:
 273   │        return keycode;
 274   │        break;
 275   │     }
 276   │ }
```

(`drivers/input/keyboard/mxc_keyb.c`)

Reply:
That seems like a rather odd choice; that exact same code could just live in user space… oh well.

Reply:
> Are there actually conflicting key binds?

as frenzie mentioned, right now only Alt+W. In future more...

Ideally I would (personally) prefer to use these as shortcuts (Alt+QW...) than the arbitrary percentage jumps (which can be found on skim widget anyway). In any case, K3 users have never had this so, technically they (we) are not really losing anything. Just my tuppence.